routes/api/v1/favorites.js

```diff
@@ -68,4 +72,51 @@ router.delete('/', (request, response) => {
     }
 });
 
+router.get('/', (request, response) => {
+    if (request.body.api_key) {
+    database('users').where('api_key', request.body.api_key).first()
+      .then((user) => {
+        if (user) {
+
+          database('favorites').where('user_id', user.id)
+            .then(favoriteLocations => {
+              if (favoriteLocations.length) {
+
+                const results = favoriteLocations.map( async (location) => {
+                    const forecastResults = await getForecasts(location, await getLatAndLongFromGoogle(location.location));
+                    return forecastResults;
+                });
+
+                Promise.all(results)
+                  .then(data => {
+                    console.log(data);
+                    response.status(200).json(data);
+                  });
+
+          } else {
+                response.status(200).json({ message: 'No favorites found.' });
+              }
+            })
+            .catch(error => {
+              response.status(500).json(error); // Sometimes gets kicked down here and I dont know why?
```

Comment:
I have yet to implement it but I think there are try { } blocks you can do to help get the errors where you want them

Reply:
Thank you! I will look up try blocks. It turns out the problem was in my async await calls with mapping over the favorites but that should be solved now.

Reply:
Oh okay, map gave me a lot of trouble as well, a Promise.all helped me there, but I suppose await too !